### PR TITLE
overlord/snapstate: SetSnapState() needs locks

### PR DESCRIPTION
**Pull request by mvo5**

### CODE REVIEW

File: overlord/snapstate/snapmgr.go [hunk 1]
```diff
@@ -157,7 +157,9 @@ func (m *SnapManager) undoPrepareSnap(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 	snapst.Candidate = nil
+	st.Lock()
```
Comment (zyga) [inline overlord/snapstate/snapmgr.go]: Perhaps we could just hold one lock during the whole call to this function?

Comment (niemeyer) [inline overlord/snapstate/snapmgr.go]: Yes, that's a better approach. We're only fiddling with state here really. Let's just defer st.Unlock.

**Comment by mvo5** [comment]
Thanks, good feedback!

**Comment by zyga** [comment]
+1

**Pull request merged**